tests/unit/acquisition/test_sampler.py

```diff
@@ -0,0 +1,265 @@
+# Copyright 2020 The Trieste Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import annotations
+
+import math
+
+import numpy.testing as npt
+import pytest
+import tensorflow as tf
+import tensorflow_probability as tfp
+
+from tests.util.misc import TF_DEBUGGING_ERROR_TYPES, ShapeLike, quadratic, random_seed
+from tests.util.model import GaussianProcess, QuadraticMeanAndRBFKernel, rbf
+from trieste.acquisition.sampler import (
+    BatchReparametrizationSampler,
+    DiscreteThompsonSampler,
+    GumbelSampler,
+    IndependentReparametrizationSampler,
+)
+from trieste.data import Dataset
+from trieste.space import Box
+from trieste.utils.objectives import branin
+
+
+@pytest.mark.parametrize("sample_size", [0, -2])
+def test_gumbel_sampler_raises_for_invalid_sample_size(
```

Comment:
Fine as is, but FYI it's possible to put related tests in a class for code clarity. For example, you could put all the gumbel sampler tests in a class TestGumpelSampler.